### PR TITLE
Replace `tmpdir` with `tmp_path`

### PR DESCRIPTION
**Pull request by harupy**
## Related Issues/PRs

 #xxx

## What changes are proposed in this pull request?

Replace `tmpdir` fixture (lebacy) with `tmp_path` fixture (preferred).

## How is this patch tested?

- [x] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests (describe details, including test results, below)

## Does this PR change the documentation?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: tests/catboost/test_catboost_model_export.py [hunk 1]
```diff
@@ -78,13 +78,13 @@ def reg_model():
 
 
 @pytest.fixture
-def model_path(tmpdir):
-    return tmpdir.join("model").strpath
+def model_path(tmp_path):
+    return os.path.join(tmp_path, "model")
```
Comment (harupy) [inline tests/catboost/test_catboost_model_export.py:82]: This isn't ideal, but using a `Path` object might break tests.

File: tests/autologging/test_autologging_behaviors_integration.py [hunk 2]
```diff
@@ -238,10 +238,10 @@ def mlflow_autolog():
 
 
 @pytest.mark.usefixtures(test_mode_off.__name__)
-def test_autolog_respects_silent_mode(tmpdir):
+def test_autolog_respects_silent_mode(tmp_path):
```
Comment (harupy) [inline tests/autologging/test_autologging_behaviors_integration.py:241]: There are more `tmpdir`. This PR just fixed easy-to-replace ones.

File: tests/spark/test_spark_model_export.py [hunk 3]
```diff
@@ -160,15 +160,15 @@ def spark_model_estimator(iris_df, spark_context):
 
 
 @pytest.fixture
-def model_path(tmpdir):
-    return str(tmpdir.mkdir("model"))
+def model_path(tmp_path):
+    return os.path.join(tmp_path, "model")
```
Comment (serena-ruan) [inline tests/spark/test_spark_model_export.py:164]: Will this directory exist directly?

Comment (harupy) [reply tests/spark/test_spark_model_export.py:164]: Good catch, but `save_model` should work when the directory doesn't exist.

File: tests/store/tracking/test_sqlalchemy_store_schema.py [hunk 4]
```diff
@@ -64,8 +64,9 @@ def expected_schema_file():
 
 
 @pytest.fixture()
-def db_url(tmpdir):
-    return "sqlite:///%s" % tmpdir.join("db_file").strpath
+def db_url(tmp_path):
+    db_file = tmp_path.joinpath("db_file")
+    return f"sqlite:///{db_file}"
```
Comment (serena-ruan) [inline tests/store/tracking/test_sqlalchemy_store_schema.py:69]:
```suggestion
    return "sqlite:///" + tmp_path.joinpath("db_file")
```

File: tests/projects/test_projects.py [hunk 5]
```diff
@@ -217,7 +217,7 @@ def test_run(use_start_run):
     assert tags[MLFLOW_PROJECT_ENTRY_POINT] == "test_tracking"
 
 
-def test_run_with_parent(tmpdir):  # pylint: disable=unused-argument
+def test_run_with_parent():
```
Comment (serena-ruan) [inline tests/projects/test_projects.py:220]: Why did we need it previously?

File: tests/store/tracking/test_sqlalchemy_store_schema.py (hunk 4 above)
Comment (harupy) [reply tests/store/tracking/test_sqlalchemy_store_schema.py:69]: `str` and `Path` can't be joined.

```python
>>> from pathlib import Path
>>> "a" + Path("a")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: can only concatenate str (not "PosixPath") to str
```

File: tests/projects/test_projects.py (hunk 5 above)
Comment (harupy) [reply tests/projects/test_projects.py:220]: I have zero idea.

**Review by serena-ruan — APPROVED** [review]
LGTM!

File: tests/store/tracking/test_sqlalchemy_store_schema.py (hunk 4 above)
Comment (harupy) [reply tests/store/tracking/test_sqlalchemy_store_schema.py:69]: That should also work :)

File: tests/projects/test_projects.py (hunk 5 above)
Comment (serena-ruan) [reply tests/projects/test_projects.py:220]: Yea makes sense

File: tests/store/tracking/test_sqlalchemy_store_schema.py (hunk 4 above)
Comment (harupy) [reply tests/store/tracking/test_sqlalchemy_store_schema.py:69]: Can file a follow-up if necessary

**Pull request merged**